File: src/main/java/com/example/demo/ApiController.java
package com.example.demo;

import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import tbs.framework.auth.annotations.PermissionValidated;

/**
 * @author Abstergo
 */
@RestController
@RequestMapping("/api")
public class ApiController {

    @DbPermission
    @PermissionValidated(value = "70")
    @PermissionValidated(value = "66")
    @RequestMapping("a")
    public String a() {

        return "Hello World a";
    }

    @RequestMapping("b")
    @PermissionValidated("PASS")
    public String b() {
        return "Hello World b";
    }

    @RequestMapping("c")
    @PermissionValidated("NOT PASS")
    public String c() {
        return "Hello World c";
    }

}

File: src/main/java/com/example/demo/ApiController.kt
package com.example.demo

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import tbs.framework.auth.annotations.Debounce

/**
 * @author Abstergo
 */
@RestController
@RequestMapping("/api")
class ApiController {
    @DbPermission("70", "66")
    @GetMapping("a")
    @Debounce
    fun a(): String {
        return "Hello World a"
    }

    @DbPermission("70", "66")
    @GetMapping("b")
    @Debounce
    fun b(p: Int): String {
        return "Hello World a" + p;
    }


//    @Resource
//    lateinit var aesTokenDebounce: AESTokenDebounce;
//
//    @GetMapping("getDebounce")
//    fun getDebounce(url: String): String {
//        return aesTokenDebounce.generateToken(
//            AESTokenDebounce.DebounceInfo(
//                RuntimeData.getInstance().userModel.userId,
//                url
//            )
//        );
//    }

}
